### PR TITLE
feat: Order alias target

### LINKED ISSUE #3196: Order alias targeting
Add support for targeting aliased fields in orderings. This does not include aggregates.

### PR DESCRIPTION
**Pull request by nasdf**
## Relevant issue(s)

Resolves #3196

## Description

This PR adds alias targeting to query orderings.

**Aggregate targets are not included in this PR as they require more changes.**

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

Added integration tests.

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

File: tests/integration/query/simple/with_order_test.go [hunk 1]
```diff
@@ -451,3 +451,380 @@ func TestQuerySimple_WithMultipleOrderFields_ReturnsError(t *testing.T) {
...
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "Alice",
+					"Age": 19
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(order: {_alias: {Age: ASC}}) {
```
Comment (fredcarle) [inline tests/integration/query/simple/with_order_test.go:547]: thought: It would be nice to see a test with a compound order query that includes an alias.

**Review by fredcarle — APPROVED** [review]
LGTM! Just one thought (non-blocking).

File: tests/integration/query/simple/with_order_test.go (hunk 1 above)
Comment (nasdf) [reply tests/integration/query/simple/with_order_test.go:547]: added a compound test

**Comment by islamaliev** [comment]
bug bash result: ran all tests from integration tests and many other related tests. Everything is good.

**Pull request merged**